.clang-tidy

```diff
@@ -38,30 +36,47 @@ Checks: >
     readability-*,
     -modernize-avoid-c-arrays,
     -modernize-use-bool-literals,
-    -modernize-use-auto
+    -modernize-use-auto,
+    -modernize-use-trailing-return-type,
+    -readability-magic-numbers,
+    -readability-implicit-bool-conversion,
+    -modernize-return-braced-init-list,
+    -readability-named-parameter,
```

Comment:
Do we use functions with unnamed arguments?

Reply:
Yes, in the case of unused parameters we remove the name to stop the compiler complaining.

Reply:
Ah yes right...